lib/ignored-paths.js

```diff
@@ -81,64 +166,99 @@ function findIgnoreFile(cwd) {
  * IgnoredPaths
  * @constructor
  * @class IgnoredPaths
- * @param {Array} patterns to be matched against file paths
+ * @param {Object} options object containing 'ignore', 'ignorePath' and 'patterns' properties
  */
-function IgnoredPaths(patterns) {
-    this.patterns = patterns;
-}
+function IgnoredPaths(options) {
 
-/**
- * IgnoredPaths initializer
- * @param {Object} options object containing 'ignore' and 'ignorePath' properties
- * @returns {IgnoredPaths} object, with patterns loaded from the ignore file
- */
-IgnoredPaths.load = function(options) {
-    var patterns;
+    options = mergeDefaultOptions(options);
+    this.defaultPatterns = DEFAULT_IGNORE_PATTERNS.concat(options.patterns || []);
+    this.baseDir = ".";
 
-    options = options || {};
+    this.ig = {
+        custom: new ignore.Ignore({
+            twoGlobstars: true,
+            ignore: []
+        }),
+        default: new ignore.Ignore({
+            twoGlobstars: true,
+            ignore: []
+        })
+    };
 
-    if (options.ignore) {
-        patterns = loadIgnoreFile(options.ignorePath || findIgnoreFile(options.cwd));
-    } else {
-        patterns = [];
+    if (options.dotfiles !== true) {
+        addPattern(this.ig.default, ".*");
     }
 
-    if (options.ignorePattern) {
-        patterns = patterns.concat(options.ignorePattern);
+    if (options.dotfiles !== true) {
+        addPattern(this.ig.default, ".*");
     }
 
-    return new IgnoredPaths(patterns);
-};
+    if (options.ignore !== false) {
+
+        addPattern(this.ig.default, this.defaultPatterns);
+
+        var ignorePath;
+
+        if (options.ignorePattern) {
+            addPattern(this.ig.custom, options.ignorePattern);
+        }
+
+        options.ignorePath = (options.ignorePath === "false") ? false : options.ignorePath;
```

Comment:
ignorePath set to `false` will result in no `.eslintignore` file being loaded. That's necessary in case you want to only use the patterns from cli flags while being in a directory with `.eslintignore` file. 

see the [Unit Test](https://github.com/eslint/eslint/pull/3948/files#diff-e0207692da5bc610c18bec3dbc54481dR128)